source/common/stats/scope_prefixer.cc

```diff
@@ -0,0 +1,48 @@
+#include "common/stats/scope_prefixer.h"
+
+#include "envoy/stats/scope.h"
+
+#include "common/stats/symbol_table_impl.h"
+#include "common/stats/utility.h"
+
+namespace Envoy {
+namespace Stats {
+
+// Variant of ScopePrefixer that owns the scope being prefixed, and will
+// delete it upon destruction.
+ScopePrefixer::ScopePrefixer(absl::string_view prefix, Scope* scope)
+    : prefix_(Utility::sanitizeStatsName(prefix)), scope_(scope), owns_scope_(true) {}
+
+// Variant of ScopePrefixer that references the scope being prefixed, but does
+// not own it.
+ScopePrefixer::ScopePrefixer(absl::string_view prefix, Scope& scope)
```

Comment:
I don't see this variant used, is it used in a follow up?

Also, somewhat nitty, but for clarity I would prefer a single constructor that looks like:
```
ScopePrefixer::ScopePrefixer(absl::string_view prefix, Scope& scope, bool take_ownership)
```

And then personally I would have the following members:
```
std::unique_ptr<Scope> owned_scope_;
Scope* scope_to_use_;
```
And avoid the manual deletion. WDYT?

Reply:
Yeah I like your pattern better. However I also can't justify the speculative generality here; I needed it in some past iteration and don't see a user-case now. If I have to add it back in the future I'll use your pattern.

update coming.